### PR TITLE
[Help Wanted] Port Planning interface to ROS2 (except tests)

### PR DESCRIPTION
**Pull request by ddengster**
### Description

Siphoned from PR #187 
Depends on the move_group PR #217.

There's still a number of things calls that need to be ported over. Here's the todo list:
- `spin_until_future_complete` calls
- Fix the tests
- cleanup
- Wait for move_group PR #217 and rebase master when it goes in. Otherwise, rebase on the `ddengster:move_group` branch which this branch is built on.

You should consider commits from 52c3f64 and beyond, and look at files in the `move_ros/planning/planning_interface` folder

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)

### CODE REVIEW

**Comment by JafarAbdi** [comment]
Rebase on https://github.com/ros-planning/moveit2/pull/217 and cleaning the PR commits

File: moveit_ros/planning_interface/move_group_interface/CMakeLists.txt [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_PYTHON_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_PYTHON_DESTINATION}
-)
+#install(TARGETS ${MOVEIT_LIB_NAME}_python
+#  ARCHIVE DESTINATION bin
+#  LIBRARY DESTINATION bin
+#  RUNTIME DESTINATION bin)
 
-install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION})
+install(DIRECTORY include/ DESTINATION move_group_interface/include)
```
Comment (JafarAbdi) [inline moveit_ros/planning_interface/move_group_interface/CMakeLists.txt:38]:
```suggestion
install(DIRECTORY include/ DESTINATION include)
```

File: moveit_ros/planning_interface/move_group_interface/CMakeLists.txt [hunk 2]
```diff
@@ -1,33 +1,41 @@
 set(MOVEIT_LIB_NAME moveit_move_group_interface)
 
-add_library(${MOVEIT_LIB_NAME} src/move_group_interface.cpp)
+add_library(${MOVEIT_LIB_NAME} SHARED src/move_group_interface.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(${MOVEIT_LIB_NAME} moveit_common_planning_interface_objects moveit_planning_scene_interface ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
+target_link_libraries(${MOVEIT_LIB_NAME} moveit_common_planning_interface_objects moveit_planning_scene_interface)
+#add_dependencies(${MOVEIT_LIB_NAME})
```
Comment (JafarAbdi) [inline moveit_ros/planning_interface/move_group_interface/CMakeLists.txt:6]:
```suggestion
```

File: moveit_ros/planning_interface/CMakeLists.txt [hunk 3]
```diff
@@ -87,7 +87,7 @@ include_directories(${THIS_PACKAGE_INCLUDE_DIRS})
 # add_subdirectory(py_bindings_tools)
 add_subdirectory(common_planning_interface_objects)
 add_subdirectory(planning_scene_interface)
-# add_subdirectory(move_group_interface)
+add_subdirectory(move_group_interface)
 # add_subdirectory(robot_interface)
 # add_subdirectory(test)
 add_subdirectory(moveit_cpp)
```
Comment (JafarAbdi) [inline moveit_ros/planning_interface/CMakeLists.txt:93]: ament_export_include_directories(include) -> ament_export_include_directories(${THIS_PACKAGE_INCLUDE_DIRS})

**Comment by henningkayser** [comment]
Closing, as contributions have been merged with #272

**Pull request closed without merging**